app/main.py

```diff
@@ -1 +1,29 @@
-# write your code here
+import json
+from decimal import Decimal
+
+
+def calculate_profit(filename: str) -> None:
+    with open(filename, "r") as input_file:
+        loading = json.load(input_file)
+        info = {"bought1": 0, "sold1": 0, "earned_bought": 0, "earned_sold": 0}
+        for element in loading:
+            buy = element["bought"]
+            price = element["matecoin_price"]
+            if buy is not None:
+                info["bought1"] += Decimal(buy)
+                info["earned_bought"] += Decimal(buy) * Decimal(price)
+
+            sell = element["sold"]
+            if sell is not None:
+                info["sold1"] += Decimal(sell)
+                info["earned_sold"] += Decimal(sell) * Decimal(price)
+
+        progres = Decimal(info["earned_sold"]) - Decimal(info["earned_bought"])
+        meta = Decimal(info["bought1"]) - Decimal(info["sold1"])
+        res = {
+            "earned_money": str(progres),
+            "matecoin_account": str(meta)
+        }
```

Comment:
As a general rule, do not overload the context manager with operations outside of file opening in this case. You could decrease indentation by one step for this code, it would be considered a better practice.